### DISCUSSION TITLE
Running an alternate EVM using Anvil

### DISCUSSION THREAD

**Discussion by sisyphusSmiling** [foundry-rs/foundry#9217] [Q&A] [answered]
Hello all!

I'm wondering if it's possible to run an alternate EVM locally using `anvil`. This is asked in the interest of running this [`go-ethereum` fork ](https://github.com/onflow/go-ethereum) which is EVM-equivalent plus some added functionality accessible via [project-specific precompiles](https://developers.flow.com/evm/how-it-works#precompiled-contracts) - e.g. onchain VRF.

Foundry's test suite is workable as-is in combination with mocking responses to these precompiles. However, local development using Foundry alone is an issue as the precompiles are of course specific to the forked implementation.

If this is not currently possible, is there something the project should look into doing to make this possible or is targeting an alternative EVM implementation just a no-go altogether without a custom foundry fork?

Thank you!

**Comment by onbjerg** [comment, marked as answer]
This is not possible and we have no plans to support custom EVMs in the sense you are asking for here, but we are interested in supporting some customization on top (e.g. custom precompiles). For now, it is somewhat possible to add custom precompiles if you wrap anvil, see e.g. https://github.com/foundry-rs/foundry/pull/7589
